hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,181 @@
+Note that this figure shows two channels, 0 and 1, where the second channel has a significant phase delay, such that the output pulse is high when `phase_ctr` overflows to zero.
+
+{{< wavejson >}}
+{signal: [
+  {name: 'PWM_EN', wave: '01...|....|......'},
+  {name: 'phase_ctr', wave: '2.222|2222|222222', data: ['0', '1', '2', '', '1024', '1025', '1026', '', '65533', '65534' ,'65535', '0', '1', '2', '3', '4']},
+  {name: 'PHASE_DELAY_0', wave: '2....|....|......', data: '0'},
+  {name: 'DUTY_CYCLE_0', wave: '2....|....|......', data: '1023'},
+  {name: 'pwm_out[0]', wave: '0.1..|.0..|....1.'},
+  {name: 'PHASE_DELAY_1', wave: '2....|....|......', data: '65533'},
+  {name: 'DUTY_CYCLE_1', wave: '2....|....|......', data: '3'},
+  {name: 'pwm_out[1]', wave: '01.0.|....|.1...0'}
+  ],
+ config:{skin:'narrow', hscale:2}
+}
+{{< /wavejson >}}
+
+#### Hardware-Controlled Blink Feature
+
+In addition, to the PWM enable function there is also a hardware-controlled blink feature to periodically hold the output low for many PWM cycles.
+The blink feature creates a blink-control signal, which modulates the PWM output.
+If hardware-controlled blinking is enabled, the PWM pulses are suppressed whenever the blink-control signal is low.
+
+The blink rate can be controlled to be as fast as once every 2 PWM cycles or as slow as once every 65536 PWM cycles.
+This rate is controlled by the combination of two registers {{< regref "BLINK_PARAM_0.RATE_0" >}} and {{< regref "BLINK_PARAM_0.PREDIV_0" >}}.
+The {{< regref "BLINK_PARAM_0.RATE_0" >}} register determines the period of the blink-control signal was well as the resolution in blink phase and duty-cycle, with a minimum phase resolution ranging from 1/256 to 1/2 of a cycle.
+The {{< regref "BLINK_PARAM_0.PREDIV_0" >}} is a frequency pre-divider, and can be used to further slow down the blink pattern, with no effect on the phase resolution.
+The overall blink-rate is given by:
+$$f_\textrm{blink}=\frac{f_\textrm{PWM}}{(\textrm{RATE}+1)(\textrm{PREDIV}+1)}$$
+The minimum value for {{< regref "BLINK_PARAM_0.RATE_0" >}} is 1.
+A {{< regref "BLINK_PARAM_0.RATE_0" >}} value of 0 disables the blink functionality, leaving the PWM pulses uninterrupted.
+
+The blink-control signal is controlled by an eight-bit blink counter within each PWM FSM.
+The blink counter increments once every ({{< regref "BLINK_PARAM_0.PREDIV_0" >}}+1) PWM cycles, whenever the phase counter is zero.
+The blink counter resets to zero when it reaches ({{< regref "BLINK_PARAM_0.RATE_0" >}}+1).
+
+The blink control signal rises whenever the blink counter reaches {{< regref BLINK_PARAM_0.DELAY_0 >}}, and falls whenever the blink counter reaches ({{<regref BLINK_PARAM_0.DELAY_0 >}} + {{<regref BLINK_PARAM_0.DUTY_CYCLE_0>}} + 1) (mod {{< regref BLINK_PARAM_0.RATE_0 >}} + 1).
+
+Changes to the blink rate and pre-divide registers only take effect when the blinking is disabled,
```

Comment:
Comma at end of sentence

Reply:
(This sentence has been removed in the most recent rewrite)